WordPress/openverse-api
# Clean up ES index mappings

## Description


This PR
- changes the ES mapping mode from dynamic to <s>strict</s> false (this means unmapped fields are stored but not indexed <s>this prevents adding documents with fields other than those defined in the mapping schema</s>)
- removes unnecessary schema mappings, changing them to purely stored fields
- converts some fields from text+keyword to just keyword
- moves hardcoded dictionary to external JSON files

## Testing Instructions

Since this is a change that makes the ES mappings stricter, passing CI should be adequate.

Additionally, you can do this.

1. On `main`, see the initial mappings by visiting the following URLs:
  - http://localhost:9200/image/_mapping
  - http://localhost:9200/audio/_mapping
2. Check out the PR, drop everything `just down -v` and reinitialise with `just init`.
3. Check the mappings again by visiting the above URLs and compare against the old value. Observe the above mentioned changes in the mapping fields.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
